### ISSUE TITLE
[Bug] Notification popup can't be hidden using Escape if an integrated terminal is focused

### ISSUE THREAD

**Issue by wenfangdu** [microsoft/vscode#131380]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: v1.60.0-insider
- OS Version: Windows 10

Steps to Reproduce:

1. Use `notifications.showList` command.
2. Focus integrated terminal.
3. Press <kbd>Escape</kbd>.
4. Notification popup is not hidden.

**Comment by Tyriar** [comment]
This was by design since the terminal uses escape, but we can totally improve this to route the escape to notifications only when they're showing

**Comment by Tyriar** [comment]
To verify:

- Make sure escape dismisses notifications when terminal is focused
- Make sure escape works in terminals when no notifications are showing (eg. in vim)

**Comment by wenfangdu** [comment]
@Tyriar Does `notifications.hideList` need to be skipped as well? I see two command IDs when searching "hide notifications":

![image](https://user-images.githubusercontent.com/28700378/130457453-ab647f93-d7b6-4fd2-ae2e-eee761563c1e.png)

**Comment by wenfangdu** [comment]
@Tyriar And it's still reproducible after adding them to my `settings.json`:

```json
{
  "terminal.integrated.commandsToSkipShell": ["notifications.hideList", "notifications.hideToasts"]
}
```

**Comment by Tyriar** [comment]
Yeah I tried that and it didn't work for me either, but adding it builtin worked fine.
